### ISSUE TITLE
linked SVGs in title page background SVG do not work

### ISSUE THREAD

**Issue by CWempe** [asciidoctor/asciidoctor-pdf#1937]
I created a title page background SVG for my title page.
This file references other images.

Like:

```xml
    <image
       preserveAspectRatio="none"
       inkscape:svg-dpi="512"
       width="33.599998"
       height="33.599998"
       style="image-rendering:optimizeQuality"
       xlink:href="my_image.svg"
       id="image8061"
       x="23.083784"
       y="239.09735" />
```

The result does not contain `my_image.png`. 😞 

It works however then I use a PNG file as a workaround. 👍 

```xml
       xlink:href="my_image.png"
```

Of cause a SVG would be better in the long term.

Did anybody notice this, too?
Can this be fixed?

This might be related to https://github.com/asciidoctor/asciidoctor-pdf/issues/414.

**Comment by mojavelinux** [comment]
As a workaround, I would suggest putting the SVG data directly in the SVG. That way, you can still use SVG.

**Comment by CWempe** [comment]
Thanks.

I created an issue upstream.

I guess this can be closed then?!

**Comment by mojavelinux** [comment]
Yes, let's follow up there.

**Comment by CWempe** [comment]
> To my knowledge, prawn-svg does not support using an SVG file within an SVG. (And this converter doesn't have any tests for it). Somewhere in your logs you should be seeing this warning message:
> 
> ```
> Unsupported image type supplied to image tag; Prawn only supports JPG and PNG
> ```
> 
> You'll need to file this upstream at https://github.com/mogest/prawn-svg.

Do you know where I would find the logs from prawn-svg? (on Windows 10)
I would like to check why another image cannot be referenced in my SVG.
`asciidoctor-png -v` does not show any error or debug messages in this case.

**Comment by mojavelinux** [comment]
When you are running asciidoctor-pdf, the error messages from prawn-svg, so you should be seeing them right in your console. Here's what I see when I try an SVG with an SVG inside of it:

```
asciidoctor: WARNING: problem encountered in image: /path/to/svg-with-local-svg.svg; Unsupported image type supplied to image tag; Prawn only supports JPG and PNG
```

You can find the sample image we are now using in the test suite here: https://github.com/asciidoctor/asciidoctor-pdf/blob/main/spec/fixtures/svg-with-local-svg.svg

**Comment by mojavelinux** [comment]
Please ask usage questions in the community chat at https://asciidoctor.zulipchat.com. We ask that you only file an issue once you've been able to confirm that the problem is with the software and not with the configuration.

**Comment by mojavelinux** [comment]
This was finally implemented in prawn-svg! I'll definitely include it in the next major release.

**Comment by lfvjimisola** [comment]
I saw that as well. Great news. Will this ticket be reopened or shall we follow another ticket?

**Comment by mojavelinux** [comment]
I'm currently waiting on at least one fix in prawn-svg that was introduced by the previous release (the invalid base64 dependency). So we'll nee to wait until at least to 0.35.0 release to upgrade. But once that's out, then I think we can upgrade. For gem upgrades, I tend not to make a new issue (though I'm not perfectly consistent about that). Just submit a PR and we can discuss it there.

**Comment by mojavelinux** [comment]
To provide more context, I've been watching issues come in after the most recent release of prawn-svg and its clear we need to wait for another version to stabilize some of the new additions. Feel free to ping here if you see a release that I miss.
